### ISSUE TITLE
[Manager] 'Event log options' menu is active when no client connected

### ISSUE THREAD

**Issue by AenBleidd** [BOINC/boinc#3934]
**Describe the bug**
'Event log options' menu is active when no client connected and show dialog with default options enabled

**Steps To Reproduce**
1. Run BOINC Manager
2. Shutdown connected client
3. Do not connect again
4. Press 'Event log options' menu

**Expected behavior**
No dialog is shown

**System Information**
 - OS: Windows x64
 - BOINC Version: 7.16.7

**Comment by CharlieFenton** [comment]
This bug does not occur on Macintosh. When there is no connected client (including when you shut it down from the Manager), all menus except the _File_ menu are grayed out, so you cannot select the _Event log options_ menu. This should also be the behavior on Windows, but it is not. So the bug should be "Manager menus are not disabled on MS Windows when no client is connected."

**Comment by CharlieFenton** [comment]
The code that should disable all menus except the File menu is in the method `CAdvancedFrame::OnMenuOpening()` in _clientgui/AdvancedFrame.cpp_ and `CSimpleFrame::OnMenuOpening()` in _clientgui/sg_BoincSimpleFrame.cpp_.

**Comment by RichardHaselgrove** [comment]
My https://github.com/BOINC/boinc/issues/3933#issuecomment-660990210 applies here too. The Manager might have been opened free-standing with the intention of attaching to a remote client later: some menus/options are still relevant.

**Comment by CharlieFenton** [comment]
@RichardHaselgrove The referenced code keeps certain menu items enabled. From the code in _clientgui/AdvancedFrame.cpp_ and _clientgui/sg_BoincSimpleFrame.cpp_:
`            // Always enable all items in File menu or Help menu:
            // ID_LAUNCHNEWINSTANCE, ID_SELECTCOMPUTER, ID_SHUTDOWNCORECLIENT,
            // ID_CLOSEWINDOW, wxID_EXIT, ID_HELPBOINC, ID_HELPBOINCMANAGER,
            // ID_HELPBOINCWEBSITE, wxID_ABOUT
`

If additional menu times should be kept enabled, this code can be modified to accomplish that.

**Comment by CharlieFenton** [comment]
It is noteworthy that this disabling of selected menus works on MS Windows under Simple View but not Advanced View, as I mention in #3187. Disabling menus works correctly on Macintosh for both Simple View and Advanced View. I don't have a convenient way to test under Linux.

**Comment by CharlieFenton** [comment]
I agree with @RichardHaselgrove that the same change done by #3187 for `CSimpleFrame::OnMenuOpening()` in _clientgui/sg_BoincSimpleFrame.cpp_ is also needed in the method `CAdvancedFrame::OnMenuOpening()` in _clientgui/AdvancedFrame.cpp_. 

But we still need to determine why the disabling of menus in Advanced View works on Macintosh but not MS Windows, while the same code works in Simple View on both platforms.

**Comment by delta1513** [comment]
I found the culprit after some debugging. The `__WXMSW__` constant is defined and hence [line 167 of AdvancedFrame.cpp](https://github.com/BOINC/boinc/blob/master/clientgui/AdvancedFrame.cpp#L167) is not called in the pre-processor and the wx-event `CAdvancedFrame::OnMenuOpening()` is not included in the wx event table.

I don't know much about `__WXMSW__` and all I can say is that when you put L167 directly in the event table, it works as expected.

Tested on Windows 10 VS2019.
